### DISCUSSION TITLE
sentry-native `sentry_init` for multithreaded application

### DISCUSSION THREAD

**Discussion by martirosyanH** [getsentry/support#56] [Support] [answered]
Hi,

I want to integrate sentry-native in my project.
If I write 
`
	sentry_options_t* options = sentry_options_new();
	sentry_options_set_dsn(options, "myDsn");
	const auto tmp = sentry_init(options);
` 
in my main.cpp and demonstrate crash there, the dump file is created and reported to dashboard properly.

But if crash happens for example in a.cpp or b.cpp, the file is not being generated.
`a.cpp and b.cpp` is linked to my main.

I would like to know what is the proper way of initializing sentry once and handle crashes in the whole application.

Thanks in advance.

**Comment by bruno-garcia** [comment, marked as answer]
The initialization should be the first thing in your `main` function. That also means only 1 thread will be running at that point, and there's no concern with concurrent access to `init`. Then, regardless of what code runs and causes the crash, the SDK will capture it. That is regardless of what file the code that caused the crash lives in.
